File: erlang/leap/src/leap.erl
-module(leap).

-export([leap_year/1, test_version/0]).

leap_year(Year) when Year rem 400 =:= 0 -> true;
leap_year(Year) when Year rem 100 =:= 0 -> false;
leap_year(Year) when Year rem 4 =:= 0 -> true;
leap_year(_) -> false.

test_version() -> 4.

File: scala/leap/src/main/scala/Leap.scala
object Leap {
  def leapYear(year: Int): Boolean = {
    var divisibleBy = (x: Int) => year % x == 0

    if (divisibleBy(400)) true
    else if (divisibleBy(100)) false
    else if (divisibleBy(4)) true
    else false
  }
}
